cybersecurity/port-scanner/devashmit-day12/port_scanner.py

```
@@ -0,0 +1,39 @@
+    try:
+        s=socket.socket(socket.AF_INET,socket.SOCK_STREAM); s.settimeout(0.5)
+        if s.connect_ex((target,port))==0:
+            svc=SERVICES.get(port,"Unknown")
+            with lock: open_ports.append((port,svc)); print(f"  [OPEN] {port:<6} {svc}")
+        s.close()
+    except: pass
+
+def main():
+    print("\n🔐 Multi-threaded Port Scanner\n   Only scan systems you own or have permission to scan.\n")
+    target=input("Target IP: ").strip()
+    try: ip=socket.gethostbyname(target)
```

Ruff (0.15.15)

[error] Multiple statements on one line (colon)

(E701)